server/src/test/java/org/elasticsearch/index/IndexSettingsTests.java

```diff
@@ -545,4 +549,94 @@ public void testCustomDataPathDeprecated() {
         assertThat(indexSettings.hasCustomDataPath(), is(true));
         assertSettingDeprecationsAndWarnings(new Setting<?>[] { IndexMetadata.INDEX_DATA_PATH_SETTING });
     }
+
+    public void testUpdateTimeSeriesTimeRange() {
+        long startTime = Math.min(randomMillisUpToYear9999(), DateUtils.MAX_MILLIS_BEFORE_9999 - 86400000);
```

Comment:
I tend to do `TimeUnit.DAY.toMillis(1)` just so its super clear how many milliseconds it is. I always have to double check numbers like `86400000`.